lib/iris/__init__.py

```diff
@@ -224,7 +224,7 @@ def __repr__(self):
     def __setattr__(self, name, value):
         if name in self.deprecated_options:
             level = self.deprecated_options[name]
-            if level == 'error':
+            if level == 'error' and not value:
```

Comment:
I don't understand the purpose of this change. If `__setattr__` has been called, there will **always** be a `value`:

```python
>>> import iris
>>> # Specifically enable deprecated behaviour.
>>> iris.FUTURE.cell_datetime_objects = False  # key=cell_datetime_objects; value=False
DeprecationWarning...
>>>
```

This is why the test is failing, as it is testing precisely this behaviour, is being passed a value and is thus failing as this `if` block evaluates to False.

Reply:
`not False == True`...

Reply:
I should have said `not True == False` which is the case when we don't want this `if` block to run.

Reply:
Of course it is. That always catches me out.